src/services/tierValidationService.js

```
@@ -0,0 +1,529 @@
+     */
+    checkActionLimits(action, tierLimits, currentUsage, options) {
+        switch (action) {
+            case 'analysis':
+                return this.checkAnalysisLimits(tierLimits, currentUsage);
+            
+            case 'roast':
+                return this.checkRoastLimits(tierLimits, currentUsage);
+            
+            case 'platform_add':
+                const { platform } = options;
+                return this.checkPlatformLimits(tierLimits, currentUsage, platform);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)